sunpy/timeseries/sources/goes.py

```diff
@@ -115,6 +117,37 @@ def plot(self, axes=None, **kwargs):
         axes.fmt_xdata = matplotlib.dates.DateFormatter("%H:%M")
         return axes
 
+    @property
+    def satellite_number(self):
+        """
+        Parses the various metafields to extract GOES satellite number.
+        """
+        #various pattern matches for the meta fields.
+        pattern_old = ("{}go{SatelliteNumber:02d}{}{month:2d}{day:2d}.fits{}")
+        pattern_new = ("{}sci_gxrs-l2-irrad_g{SatelliteNumber:02d}_d{year:4d}{month:2d}{day:2d}_{}.nc{}")
+        pattern_r = ("{}sci_xrsf-l2-flx1s_g{SatelliteNumber:02d}_d{year:4d}{month:2d}{day:2d}_{}.nc{}")
+        pattern_telescop=("GOES {SatelliteNumber:02d}")
+        pattern_inst=("{}GOES 1-{SatelliteNumber:02d} {}")
+
+        try: 
+            id = self.meta.metas[0]['id']
+            parsed=parse(pattern_r,str(id))
+            if parsed==None:
+                parsed=parse(pattern_new,str(id))
+                if parsed==None:
+                    parsed=parse(pattern_old,str(id))
+                    if parsed==None:
+                        id=self.meta.metas[0]['Instrument']
+                        parsed=parse(pattern_inst,str(id))
+        except KeyError:
```

Comment:
What line of code raises the `KeyError`?

Reply:
133/140 for the first and 144 for second, basically anywhere there is an id =